pkg/api/v1/types.go

```diff
@@ -2301,6 +2307,10 @@ type DeleteOptions struct {
 	// specified type will be used.
 	// Defaults to a per object value if not specified. zero means delete immediately.
 	GracePeriodSeconds *int64 `json:"gracePeriodSeconds"`
+
+	// Must be fulfilled before a deletion is carried out. If not possible, a 409 Conflict status will be
+	// returned.
+	Preconditions Preconditions `json:"preconditions"`
```

Comment:
Delete options are sent as the body of the request, so it should be fine.

On Mon, Mar 21, 2016 at 12:14 PM, Brian Grant <notifications@github.com>
wrote:

> In pkg/api/v1/types.go
> <https://github.com/kubernetes/kubernetes/pull/22965#discussion_r56879709>
> :
>
> > @@ -2301,6 +2307,10 @@ type DeleteOptions struct {
> >  	// specified type will be used.
> >  	// Defaults to a per object value if not specified. zero means delete immediately.
> >  	GracePeriodSeconds *int64 `json:"gracePeriodSeconds"`
> > +
> > +	// Must be fulfilled before a deletion is carried out. If not possible, a 409 Conflict status will be
> > +	// returned.
> > +	Preconditions Preconditions `json:"preconditions"`
>
> I'm fine with the general idea, and with the name of the field.
>
> However, does a nested struct work with the automagic option converter?
>
> Otherwise, same comments as on api/types.go.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/22965/files/153666f7547ec3968df6198fe5717ca1682ecc52#r56879709>
>